### PR TITLE
 Adding option to force client auth on the server SSL creds.

### PR DESCRIPTION
**Pull request by jboeuf**
Addresses #2527. 
@jtattermusch @murgatroid99 @tbetbetbe @nathanielmanistaatgoogle should have a look at how to expose this new options for the wrapped languages.

### CODE REVIEW

File: include/grpc++/server_credentials.h [hunk 1]
```diff
@@ -64,6 +64,7 @@ struct SslServerCredentialsOptions {
   };
   grpc::string pem_root_certs;
   std::vector<PemKeyCertPair> pem_key_cert_pairs;
+  bool force_client_auth;
```
Comment (yang-g) [inline include/grpc++/server_credentials.h]: Should we add a default ctor for this? Or update test/cpp/interop/server_helper.cc to set it there?

**Comment by yang-g** [comment]
LGTM otherwise

File: include/grpc++/server_credentials.h (hunk 1 above)
Comment (yang-g) [inline include/grpc++/server_credentials.h]: If we do not add a default ctor, then all the existing uses of the SslServerCredentialsOptions will need to be looked at, right?

Comment (jboeuf) [inline include/grpc++/server_credentials.h]: Good catch, thanks! fixed.

**Pull request merged**